### PR TITLE
chore(smart-contracts): publish to tenderly

### PR DESCRIPTION
**Pull request by alexandre-abrioux**
## Description of the changes

Adds a command to import contracts in a Tenderly account

### Related
- https://app.asana.com/0/1203350802794955/1203755698846365/f
- https://app.asana.com/0/1203350802794955/1203755698846366/f

### CODE REVIEW

File: packages/smart-contracts/scripts-create2/utils/chains.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+type ChainConfig = {
+  name: string;
+  chainId: number;
+  rpcUrls: string[];
+};
+
+const cachedChainConfigs: Record<string, ChainConfig> = {};
+
+export const getChainConfig = async (chainName: string) => {
```
Comment (alexandre-abrioux) [inline packages/smart-contracts/scripts-create2/utils/chains.ts:11]: - moved this function out from the `toolbox` utils to the `smart-contracts` utils
- added a simple cache

File: packages/smart-contracts/scripts-create2/utils/utils.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { Contract } from 'ethers';
```
Comment (benjlevesque) [inline packages/smart-contracts/scripts-create2/utils/utils.ts:1]: rename this file `index`? or anything else than `utils` 😬

**Review by benjlevesque — COMMENTED** [review]
nice! 

A few comments:
- what will happen on chains not supported by tenderly? 
- could we avoid relying on the (request finance) currency API, in a core library of the protocol? It seems it's used only to get the chainId, which can probably be found with a call to the RPC, that I think is already available in Hardhat configuration?

File: packages/smart-contracts/src/lib/ContractArtifact.ts [hunk 3]
```diff
@@ -94,6 +94,24 @@ export class ContractArtifact<TContract extends Contract> {
     }));
   }
 
+  /**
+   * Retrieve all addresses for all versions for all networks
+   * @returns the addresses of the deployed contract and the associated network and version.
+   */
+  getAllAddressesFromAllNetworks(): { version: string; address: string; networkName: string }[] {
```
Comment (kevindavee) [inline packages/smart-contracts/src/lib/ContractArtifact.ts:101]: No enum or string union for the `networkName`?

**Review by kevindavee — APPROVED** [review]

File: packages/smart-contracts/src/lib/ContractArtifact.ts (hunk 3 above)
Comment (alexandre-abrioux) [reply packages/smart-contracts/src/lib/ContractArtifact.ts:101]: not sure that we have this ☹️

**Review by leoslr — APPROVED** [review]

File: packages/smart-contracts/src/lib/ContractArtifact.ts (hunk 3 above)
Comment (alexandre-abrioux) [reply packages/smart-contracts/src/lib/ContractArtifact.ts]: fixed in https://github.com/RequestNetwork/requestNetwork/pull/1056

**Review by benjlevesque — APPROVED** [review]

**Review by KolevDarko — APPROVED** [review]

**Pull request merged**